### PR TITLE
VZ-2603: STaaS issue

### PR DESCRIPTION
**Pull request by kminder**
# Description

Prevents Istio Envoy from generating HTTP Server header.  Also adds a test for the ingress-nginx which is currently disabled but can be enabled when VZ-2599 is completed.

Fixes VZ-2603: STaaS issue

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [x] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: tests/e2e/examples/helidon/helidon_example_test.go [hunk 1]
```diff
@@ -177,8 +179,37 @@ func helloHelidonPodsRunning() bool {
...
+	}
+	req.Host = hostname
+	httpClient := pkg.GetVerrazzanoHTTPClient()
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		pkg.Log(pkg.Error, fmt.Sprintf("Unexpected error=%v", err))
+		return false
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
```
Comment (pfmackin) [inline tests/e2e/examples/helidon/helidon_example_test.go:195]: You have to always read the body or there will be handle leak.   See https://github.com/golang/go/blob/master/src/net/http/response.go. Response.Body comments.

File: platform-operator/thirdparty/charts/istio/gateways/istio-ingress/templates/preconfigured.yaml [hunk 2]
```diff
@@ -70,3 +70,26 @@ spec:
       mode: ISTIO_MUTUAL
 ---
 {{- end }}
+## Prevent generation of HTTP Server header.
```
Comment (bobdonat) [inline platform-operator/thirdparty/charts/istio/gateways/istio-ingress/templates/preconfigured.yaml:73]: My understanding is we should not be changing the thirdparty helm charts.  Can this be applied in the istio install script instead.

Comment (kminder) [reply platform-operator/thirdparty/charts/istio/gateways/istio-ingress/templates/preconfigured.yaml:73]: This is a good point.  For some reason I saw the multi-cluster stuff and thought that was "our stuff".  I'll figure out a different way to apply this.

File: tests/e2e/examples/helidon/helidon_example_test.go (hunk 1 above)
Comment (kminder) [reply tests/e2e/examples/helidon/helidon_example_test.go:195]: Ok, I'll fix.

**Review by pfmackin — APPROVED** [review]

**Pull request merged**